### DISCUSSION TITLE
Problem with Select Multiple on datagrid and filtering too

### DISCUSSION THREAD

**Discussion by emmanouela88** [Megabit/Blazorise#4228] [Q&A] [answered]
Hello,

I am not sure if this question has been answered before (I have been through most of the Q&A's that have either of the specific words but I came empty)but, I have a datagrid and I want on a specific column to have a custom made filter. I am using the <FilterTemplate> component as you have also on your documentation but this is the problem that I am facing. I am trying to have a multiple selection like the autocomplete component but I don't want to have a text input and also the filtering is not working when I am selecting one of the entries.
...

```cshtml
<DataGridColumn Field="@nameof(MyClass.MyMethod)" Caption="My Methods" CustomFilter="@OnSelectedMethod">
<FilterTemplate>
                <Select TValue="string" SelectedValue="@selectedMethod" SelectedValueChanged="@(value=>{selectedMethod=value;context.TriggerFilterChange(selectedMethod);})" Multiple>
                    <SelectItem Value="@("*")">All</SelectItem>
                    <SelectItem Value="@("Email")">Email</SelectItem>
                    <SelectItem Value="@("Viber/SMS")">Viber/SMS</SelectItem>
                    <SelectItem Value="@("SMS")">SMS</SelectItem>
                </Select>
            </FilterTemplate>
</DataGridColumn>
...
@code{
private bool OnSelectedMethod(object itemValue, object searchValue){

        if(searchValue is string methodFilter){
            return methodFilter == "*" || methodFilter.Contains(itemValue?.ToString());// == itemValue?.ToString();

        }

        return true;
    }}
```

When I am using the multiple and trying to ctrl+click nothing happens on the filtering...

I hope to get some suggestions on how to tackle this problem. Thank you in advance.

**Comment by stsrki** [comment, marked as answer]
You're using the wrong parameters. When `Multiple` is enabled, you must use `SelectedValues` and `SelectedValuesChanged`.

**Reply by emmanouela88** [reply]
First of all thanks for your quick answer!

Hmm I see what you are saying. I have tried to changed the parameteres to the correct ones but the thing is that when I am using the SelectedValuesChanged I am not sure how to initialize the corresponding method or use somehting similar to what I have written on the SelectedValueChange or as a method that you have on the documentation (for SelectedValueChange). I was trying to find something similar but with no luck. Is there a way to provide me with an example?

**Reply by stsrki** [reply]
As long as the value you provide is of `IReadOnlyList<TValue>` that it should work.

**Reply by emmanouela88** [reply]
Oh okay.

Thank you very much for your help :D
